src/linear_algebra/alternating.lean

```diff
@@ -434,3 +444,292 @@ lemma comp_multilinear_map_alternatization (g : N' →ₗ[R] N'₂)
 by { ext, simp [multilinear_map.alternatization_def] }
 
 end linear_map
+
+section coprod
+
+open_locale big_operators
+open_locale tensor_product
+
+variables {ιa ιb : Type*} [decidable_eq ιa] [decidable_eq ιb] [fintype ιa] [fintype ιb]
+
+namespace equiv.perm
+
+/-- Elements which are considered equivalent if they differ only by swaps within α or β  -/
+abbreviation mod_sum_congr (α β : Type*) :=
+quotient_group.quotient (equiv.perm.sum_congr_hom α β).range
+
+end equiv.perm
+
+namespace alternating_map
+open equiv
+
+/-- Auxiliary helper for `dom_coprod`, before it has been proven that the result is alternating. -/
+def dom_coprod_fun
+  {R : Type*} {M N₁ N₂ : Type*}
+  [comm_semiring R]
+  [add_comm_group N₁] [semimodule R N₁]
+  [add_comm_group N₂] [semimodule R N₂]
+  [add_comm_monoid M] [semimodule R M]
```

Comment:
I would use `variables` for this.

Reply:
I didn't do this because there is already a global `R` variable in the file with an associated `semiring`. It's frustrating that `def ... {R : Type*}` can shadow variables, but `section` `variables {R : Type*}` `def ...` can't.

I guess I'll just introduce an `R'` and use that instead...